### ISSUE TITLE
Comments from Adrian about APS

### ISSUE THREAD

**Issue by italobusi** [tsaad-dev/te#216]
> > Why does this text talk about the APS state machine when 8776 makes
> > reference to 4427? Possibly the references in 8776 are wrong and should
> > have pointed to 7271 and 8234. If you intend those meanings of APS then
> > you should simply add references to your document.
> > 
> > (Raising an Errata Report against 8776 for this would be a bonus)
> > 
> [TS]: Added reference to RFC7271 and RFC8234.
> [TS/OPEN-ISSUE]: to track other suggestions.
> 

[Italo Busi] RFC7271 and RFC8234 are applicable to MPLS-TP only (MPLS-TP PSC protocol operations in APS mode). For the generic definition of APS we might reference ITU-T G.808 (we thought RFC4427 was aligned but Adrian is right that RFC4427 is not covering APS definitions).

> > 5.3
> > 
> > I think the reference for aps-signal-id is wrong.
> 
> [TS/OPEN-ISSUE]: working with team to address this.
> 

[Italo Busi] I think we can reference clause 16 of ITU-T G.808.1.

See: https://mailarchive.ietf.org/arch/msg/teas/_aZOPSSjMychndBfCRjX0HAWXnA/

**Comment by italobusi** [comment]
Regarding the first comment from Adrian:

> > Why does this text talk about the APS state machine when 8776 makes
> > reference to 4427? Possibly the references in 8776 are wrong and should
> > have pointed to 7271 and 8234. If you intend those meanings of APS then
> > you should simply add references to your document.
> > 
> > (Raising an Errata Report against 8776 for this would be a bonus)
> > 

Checking the current definition of the identity `lsp-protection-state` in RFC8776, I have noted that there is no mention of APS nor any reference:

```
  identity lsp-protection-state {
    description
      "Base identity of protection states for reporting purposes.";
  }
```

I am now wondering whether we really need to need to mention APS here, and add a reference in [draft-ietf-teas-yang-te](https://datatracker.ietf.org/doc/html/draft-ietf-teas-yang-te) and in RFC8776-bis.

What about about the following change:

OLD

```
       leaf lsp-protection-state {
         type identityref {
           base te-types:lsp-protection-state;
         }
         default "te-types:normal";
         description
           "The state of the APS state machine controlling which
            tunnels is using the resources of the protecting LSP.";
       }
```

NEW

```
       leaf lsp-protection-state {
         type identityref {
           base te-types:lsp-protection-state;
         }
         config false;
         description
           "The reported protection state controlling which
            tunnels is using the resources of the protecting LSP.";
       }
```

---

Regarding the second comment from Adrian:

> > 5.3
> > 
> > I think the reference for aps-signal-id is wrong.
> 

I would suggest the following change in YANG (see also updated PR #205):

OLD
```
      leaf aps-signal-id {
        ...
        reference
          "RFC4427";
      }
    }
```
NEW
```
      leaf aps-signal-id {
        ...
        reference
          "ITU-T G.808.1 (05/2014)";
      }
    }

```

Note: since it is a report, it shall be config false with no default as per common practice in this model

I would suggest the following changes to the md file (see also updated PR #205: https://github.com/tsaad-dev/te/pull/205/commits/b67f9d3ba936af39b06bd6f91232a56ce7ba7f18):

OLD
```
<...>
  RFC8040:

informative:
<...>
This module references the following documents:
<...>
{{!RFC8232}}, {{!RFC7271}}, {{!RFC8234}}, and {{!RFC7308}}.
```
NEW
```
<...>
  RFC8040:
  ITU_G.808.1:
    title: Generic protection switching - Linear trail and subnetwork protection
    author:
      org: ITU-T Recommendation G.808.1
    date: May 2014
    seriesinfo: ITU-T G.808.1

informative:
<...>
This module references the following documents:
<...>
{{!RFC8232}}, {{!RFC7271}}, {{!RFC8234}}, {{!RFC7308}}, and {{ITU_G.808.1}}.
```

**Comment by tsaad-dev** [comment]
Addressed in 902c1ad

**Comment by italobusi** [comment]
It seems that merging two comments in the same issue has been a bit confusion (they are related but slightly different), so I have split the discussion into two issues: #224  and #225
